src/helper/bus.inputs.js

```diff
@@ -0,0 +1,42 @@
+import Joi from '@hapi/joi';
+
+const busInput = (req) => {
+  const schema = Joi.object().keys({
+    routId: Joi.number().integer().required().min(1)
+      .max(150),
+    bus_plate: Joi.string().required().min(5),
+    currentLocation: Joi.string().required().custom((value, helper) => {
+      const coordinates = req.body.currentLocation.split(',');
+      if (!req.body.currentLocation.includes(',')) {
+        return helper.message('Latitude and Longitude must be separated by comma');
+      }
+      const lat = coordinates[0].trim();
+      const long = coordinates[1].trim();
+      const valRegex = /-?\d/;
+      if (!valRegex.test(lat)) {
+        return helper.message('Laltitude must be numbers');
+      }
+      if (!valRegex.test(long)) {
+        return helper.message('Longitude must be numbers');
+      }
+    }),
+    bus_status: Joi.string().required().valid('active', 'inactive'),
```

Comment:
Actually, bus_status should not be required. If not provided, it should default to inactive

Reply:
Yes, done